### ISSUE TITLE
The example of get() in "Readable cdb" is not working

### ISSUE THREAD

**Issue by Currerius** [ericnorris/node-cdb#1]
Hi,
I am new to js, node and cdb. I started writing my first small scripts and want to read from a cdb.

When pasting your small examples for writable and readable into say "WritableCDB.js" and "ReadableCDB.js" and executing them in the shell with "node Writable.js" and "node Readable.js" the former is working correctly and the cdbfile is correctly created - I validated the file with freecdb.

But:

```
node ReadableCDB.js
```

returns "undefined" and not "hello world".

I changed "console.log(data)" to "console.log(err,data)" and the error code is

```
[Error: EBADF, read] errno: 9, code: 'EBADF'
```

I did some search on this one and it seems that close() is called before the data is read. I have not understood the callback structure good enough to know how to correct it, but commenting out the close() line returns the data.

But: in hex encoding :)

So:

```
reader.open(function cdbOpened(err) {
    reader.get('meow', function gotRecord(err, data) {
        console.log( data.toString() );
    });
});
```

Produces "hello world", but then the connection is not closed.
Would be curious how to get the callback right.

Thanks,
Reinhard

**Comment by Currerius** [comment]
... when I change the "meow" key to "meo" (not matching) the error disappears and "null" is correctly returned...

**Comment by Currerius** [comment]
This might be correct? It is working and when it is correct I start understanding callback(). node-cdb was a useful and easy starting point. Thx.

```
var readable = require('constant-db').readable;

var reader = new readable('./cdbfile');

reader.open(function cdbOpened(err) {

    console.log('open');

    reader.get('meow', function gotRecord(err, data) {

        console.log('read');

        console.log(data.toString());

        reader.close( function cdbClosed(err) {
            console.log('close');})

    });

});
```

... the close() has to be inside the callback() of the get() it seems.

**Comment by ericnorris** [comment]
Good catch @Currerius, it appears I made a mistake in the documentation. You are right, the `close()` call needs to be inside the callback for the `read()` call, so that it happens _after_ the `read()`.

Thanks!

**Comment by dwbruhn** [comment]
The npm documentation (https://www.npmjs.com/package/constant-db) still contains this mistake, unfortunately.

**Comment by ericnorris** [comment]
Thanks @dwbruhn, I published v2.0.2 to NPM so it should be up to date.
:thumbsup:

**Comment by dwbruhn** [comment]
:+1:
